arch/cortex-m/samv71-hal/src/peripherals/watchdog.rs

```diff
@@ -0,0 +1,120 @@
+//! Implementation of HAL for Watchdog
+
+use embedded_hal::watchdog::WatchdogDisable;
+
+use crate::embedded_hal::watchdog;
+use crate::pac::WDT;
+
+/// Structure representing a watchdog
+pub struct Watchdog {
+    /// Watchdog instance
+    wdt: WDT,
+    /// Indicates whether the watchdog has been configured (or disabled).
+    ///
+    /// Note that watchdog can be configured or disabled only once.
+    /// Configuration is locked until MCU performs a hard reset.
+    /// This flag prevents it from being configured twice.
+    configured: bool,
+}
+
+/// SAFETY: Watchdog does not auto-implement Sync due to WDT structure.
+/// Since it owns WDT, and it's running in single-core environment, it's safe to share.
+unsafe impl Sync for Watchdog {}
+
+/// Structure representing Watchdog configuration.
+///
+/// Note that watchdog can be configured only once.
+/// Configuration is locked until MCU performs a hard reset.
+pub struct WatchdogConfiguration {
```

Comment:
Move to a different file maybe? `src/peripherals/watchdog/watchdog_config.rs`?
\+ We have `Config` in other places instead of `Configuration`

Reply:
I don't think it's necessary to split the code *that* much. I've renamed it though.

Reply:
I'd like to try to keep one struct per file. IMO it's really unreadable, when I want to find a struct `Foo` in `foo.rs` and I have to scroll through multiple 'helper' structs before I get to what I want to

Reply:
fair enough, i'm refactoring the structure a bit so i can do that

Reply:
Wil be done in the next MR as discussed